qutip/solver/stochastic.py

```diff
@@ -0,0 +1,754 @@
+        """
+        Measurements for each trajectories and stochastic collapse operators.
+
+        The output shape is
+            (ntraj, len(sc_ops), len(tlist)-1)
+        for homodyne detection, and
+            (ntraj, len(sc_ops), 2, len(tlist)-1)
+        for heterodyne detection.
+        """
+        return self._trajectories_attr("measurement")
+
+    @property
+    def dW(self):
+        """
+        Wiener increment for each trajectories and stochastic collapse
+        operators.
+
+        The output shape is
+            (ntraj, len(sc_ops), len(tlist)-1)
+        for homodyne detection, and
+            (ntraj, len(sc_ops), 2, len(tlist)-1)
+        for heterodyne detection.
+        """
+        return self._trajectories_attr("dW")
+
+    @property
+    def wiener_process(self):
+        """
+        Wiener processes for each trajectories and stochastic collapse
+        operators.
+
+        The output shape is
+            (ntraj, len(sc_ops), len(tlist)-1)
+        for homodyne detection, and
+            (ntraj, len(sc_ops), 2, len(tlist)-1)
+        for heterodyne detection.
+        """
+        return self._trajectories_attr("wiener_process")
+
+
+class _StochasticRHS:
+    """
+    In between object to store the stochastic system.
+
+    It store the Hamiltonian (not Liouvillian when possible), and sc_ops.
+    dims and flags are provided to be usable the the base ``Solver`` class.
+
+    We don't want to use the cython rhs (``StochasticOpenSystem``, etc.) since
+    the rouchon integrator need the part but does not use the usual drift and
+    diffusion computation.
+    """
+
+    def __init__(self, issuper, H, sc_ops, c_ops, heterodyne):
+
+        if not isinstance(H, (Qobj, QobjEvo)) or not H.isoper:
+            raise TypeError("The Hamiltonian must be am operator")
+        self.H = QobjEvo(H)
+
+        if isinstance(sc_ops, (Qobj, QobjEvo)):
+            sc_ops = [sc_ops]
+        self.sc_ops = [QobjEvo(c_op) for c_op in sc_ops]
+
+        if isinstance(c_ops, (Qobj, QobjEvo)):
+            c_ops = [c_ops]
+        self.c_ops = [QobjEvo(c_op) for c_op in c_ops]
+
+        if any(not c_op.isoper for c_op in c_ops):
+            raise TypeError("c_ops must be operators")
+
+        if any(not c_op.isoper for c_op in sc_ops):
+            raise TypeError("sc_ops must be operators")
+
+        self.issuper = issuper
+        self.heterodyne = heterodyne
+
+        if heterodyne:
+            sc_ops = []
+            for c_op in self.sc_ops:
+                sc_ops.append(c_op / np.sqrt(2))
+                sc_ops.append(c_op * (-1j / np.sqrt(2)))
+            self.sc_ops = sc_ops
+
+        if self.issuper and not self.H.issuper:
+            self.dims = [self.H.dims, self.H.dims]
+        else:
+            self.dims = self.H.dims
+
+    def __call__(self, options):
+        if self.issuper:
+            return StochasticOpenSystem(
+                self.H, self.sc_ops, self.c_ops, options.get("dt", 1.0 - 6)
```

Comment:
```suggestion
                self.H, self.sc_ops, self.c_ops, options.get("dt", 1e-6)
```
Was this intended to be `1e-6`? `-5` seems an odd default value for `dt`.